Tasks/Common/nuget-task-common/Utility.ts

```diff
@@ -82,7 +82,11 @@ export function resolveWildcardPath(pattern: string, allowEmptyWildcardMatch?: b
         // First find the most complete path without any matching patterns
         let idx = firstWildcardIndex(pattern);
         tl.debug("Index of first wildcard: " + idx);
-        let findPathRoot = path.dirname(pattern.slice(0, idx));
+
+        // include the wildcard character because:
+        //  dirname(c:\foo\bar\) => c:\foo (which will make find() return a bunch of stuff we know we'll discard)
+        //  dirname(c:\foo\bar\*) => c:\foo\bar
+        let findPathRoot = path.dirname(pattern.slice(0, idx + 1));
```

Comment:
for globbing consistency, the strategy is: 1) task author guidance (doc linked above). 2) good defaults for the glob options. changing the defaults for the glob functions was a breaking change in the 2x task lib. 3) update in the box tasks. they are open source examples.

for the UX structures, what do you have in mind? this is straying far from the PR :) you might want to shoot a mail to bryan or chris